tstest/integration/integration.go

```diff
@@ -1098,21 +1100,46 @@ func (tt *trafficTrap) ServeHTTP(w http.ResponseWriter, r *http.Request) {
 }
 
 type authURLParserWriter struct {
+	t   *testing.T
 	buf bytes.Buffer
-	fn  func(urlStr string) error
+	// Handle login URLs, and count how many times they were seen
+	authURLFn    func(urlStr string) error
+	authURLCount *atomic.Int32
+	// Handle machine approval URLs, and count how many times they were seen.
+	machineAuthURLFn    func(urlStr string) error
```

Comment:
Found myself getting confused reading this in isolation and thinking it's the auth url handler. I think we call this "approval" fairly consistently, so maybe `approvalURLFn` instead? And replace things like "machineAuth" with "approval" everywhere else too?